pandas/core/frame.py

```diff
@@ -3867,23 +3867,26 @@ def _get_value(self, index, col, takeable: bool = False) -> Scalar:
 
     def isetitem(self, loc, value) -> None:
         """
-        Set the given value in the column with position 'loc'.
+        Set the given value in the column with position `loc`.
 
-        This is a positional analogue to __setitem__.
+        This is a positional analogue to `__setitem__`.
 
         Parameters
         ----------
         loc : int or sequence of ints
+            Index position for the column.
         value : scalar or arraylike
+            Value(s) for the column.
 
         Notes
         -----
-        Unlike `frame.iloc[:, i] = value`, `frame.isetitem(loc, value)` will
-        _never_ try to set the values in place, but will always insert a new
-        array.
+        ``frame.isetimtem(loc, value)`` is an in-place method as it will
+        modify the frame in place (not returning a new object) but it will
+        not update the values of the column itself, it will instead insert
+        a new array.
```

Comment:
I would add something like "in contrast to `frame.iloc[:, i] = value` which will try to update the existing values inplace"

Reply:
I just made some examples on my side and indeed this is quite important to keep that in the docs, I will update that

Reply:
We should be good now, let me know if the phrasing is clear enough

Reply:
We should be good now, let me know if the phrasing is clear enough